lib/travis/build/script/c.rb

```diff
@@ -43,7 +43,13 @@ def use_directory_cache?
         private
 
           def compiler
-            config[:compiler].to_s
+            case config[:compiler].to_s
+            when /^gcc/i, /^clang/i
+              config[:compiler].to_s
+            else
+              { 'linux' => 'gcc', 
+                'osx' => 'clang' }[config[:os]]
+            end
           end
```

Comment:
Google taught me there is a `start_with`, so this should be a prettier version.

    if config[:compiler].to_s.start_with?('gcc', 'clang')
      config[:compiler].to_s
    else
      case config[:os]
      when 'osx'
        'clang'
      when 'linux'
        'gcc'
      end
    end

Also, at this point the blank default gets overwritten by the platform specific default. It would be even nicer and maybe necessary for the cache slug generation to move the platform-specific default generation to line 6, and then call `DEFAULTS.compiler` or something, but it isn't required for the platform announcement. And as said before, I don't know how to do that anyway :-)